From 1528bec5509d9c26c479cfd3c2188c2b34621258 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 26 Aug 2014 15:30:30 -0400
Subject: [ethernet] igbvf: remove open-coded skb_cow_head

Message-id: <1409067034-17415-10-git-send-email-sassmann@redhat.com>
Patchwork-id: 89221
O-Subject: [RHEL7.1 PATCH 09/13] igbvf: remove open-coded skb_cow_head
Bugzilla: 1091122
RH-Acked-by: Neil Horman <nhorman@redhat.com>
RH-Acked-by: John Greene <jogreene@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: John Linville <linville@redhat.com>

From: Francois Romieu <romieu@fr.zoreil.com>

Signed-off-by: Francois Romieu <romieu@fr.zoreil.com>
Cc: Jesse Brandeburg <jesse.brandeburg@intel.com>
Tested-by: Aaron Brown <aaron.f.brown@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 6b8f07b4904b3fcd17dcb91165a218266727b4cf)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/igbvf/netdev.c b/drivers/net/ethernet/intel/igbvf/netdev.c
index b7ab03a..d608599 100644
--- a/drivers/net/ethernet/intel/igbvf/netdev.c
+++ b/drivers/net/ethernet/intel/igbvf/netdev.c
@@ -1910,20 +1910,18 @@ static int igbvf_tso(struct igbvf_adapter *adapter,
                      struct sk_buff *skb, u32 tx_flags, u8 *hdr_len)
 {
  struct e1000_adv_tx_context_desc *context_desc;
- unsigned int i;
- int err;
  struct igbvf_buffer *buffer_info;
  u32 info = 0, tu_cmd = 0;
  u32 mss_l4len_idx, l4len;
+ unsigned int i;
+ int err;
+
  *hdr_len = 0;
 
- if (skb_header_cloned(skb)) {
-  err = pskb_expand_head(skb, 0, 0, GFP_ATOMIC);
-  if (err) {
-   dev_err(&adapter->pdev->dev,
-           "igbvf_tso returning an error\n");
-   return err;
-  }
+ err = skb_cow_head(skb, 0);
+ if (err < 0) {
+  dev_err(&adapter->pdev->dev, "igbvf_tso returning an error\n");
+  return err;
  }
 
  l4len = tcp_hdrlen(skb);
-- 
1.7.1